codex/merkletree/codex/codex.nim

```diff
@@ -145,16 +145,18 @@ func compress*(
   mhash.coder(@x & @y & @[ key.byte ], digest)
   success digest
 
-func init*(
+proc init*(
```

Comment:
I think we might need to avoid making this `init` async so we can avoid using `waitFor` later on... instead this is a good pattern to follow: 
https://github.com/codex-storage/nim-ethers/blob/80b2ead97ce32cca74cef4a4501ab97106cb578b/ethers/providers/jsonrpc.nim#L50-L86

Then later, we can use `CodexTree.init` synchronously, and when needed, we can `let layers = await tree.layers`